### PR TITLE
Create Deploying wallet instead of FutureWallet if refund is paid

### PR DESCRIPTION
**Pull request by Sajma27**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [ ] Change is small and easy to review (please split big changes into multiple PRs)
- [ ] Change is consistent with architecture
- [ ] Change is consistent with test architecture
- [ ] Change is consistent with naming conventions
- [ ] New code is covered with tests
- [ ] Tests related to old code are updated
- [ ] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-sdk/src/core/services/WalletService.ts [hunk 1]
```diff
@@ -67,7 +67,13 @@ export class WalletService {
     return this.state.wallet;
   }
 
-  async createFutureWallet(name: string): Promise<FutureWallet> {
+  async createFutureOrDeployingWallet(name: string): Promise<FutureWallet | DeployingWallet> {
+    if (this.sdk.isRefundPaid()) {
+      const futureWallet = await this.sdk.createFutureWallet(name, '0', ETHER_NATIVE_TOKEN.address);
+      const deployingWallet = await futureWallet.deploy();
+      this.setDeploying(deployingWallet);
+      return deployingWallet;
```
Comment (xitronix) [inline universal-login-sdk/src/core/services/WalletService.ts:75]: extract

File: universal-login-sdk/src/core/services/WalletService.ts [hunk 2]
```diff
@@ -67,7 +67,13 @@ export class WalletService {
...
+    }
     const gasModes = await this.sdk.getGasModes();
     const gasOption = findGasOption(gasModes[FAST_GAS_MODE_INDEX].gasOptions, ETHER_NATIVE_TOKEN.address);
     const futureWallet = await this.sdk.createFutureWallet(name, gasOption.gasPrice.toString(), gasOption.token.address);
```
Comment (xitronix) [inline universal-login-sdk/src/core/services/WalletService.ts:79]: extract

File: universal-login-sdk/src/core/services/WalletService.ts [hunk 3]
```diff
@@ -67,7 +67,13 @@ export class WalletService {
     return this.state.wallet;
   }
 
-  async createFutureWallet(name: string): Promise<FutureWallet> {
+  async createFutureOrDeployingWallet(name: string): Promise<FutureWallet | DeployingWallet> {
```
Comment (xitronix) [inline universal-login-sdk/src/core/services/WalletService.ts:70]: Rename

**Comment by JustynaBroniszewska** [comment]
rename `createFutureOrDeployingWallet` to `createWallet`

File: universal-login-sdk/test/core/services/WalletService.unit.test.ts [hunk 4]
```diff
@@ -113,6 +113,15 @@ describe('UNIT: WalletService', () => {
     expect(storage.set).to.be.calledWith(TEST_STORAGE_KEY, JSON.stringify({kind: 'Deployed', wallet: applicationWallet}));
   });
 
+  it('create deploying wallet if isRefundPaid', async () => {
+    expect(walletService.state).to.deep.eq({kind: 'None'});
+    sdk = {...sdk, isRefundPaid: () => true, createFutureWallet: () => futureWallet};
+    walletService = new WalletService(sdk);
+    const name = 'name.mylogin.eth';
+    const deployingWallet = await walletService.createWallet(name);
+    expect(deployingWallet).instanceOf(DeployingWallet);
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/test/core/services/WalletService.unit.test.ts:122]: We can add a sanity check `expect(walletService.state).to.be(DeployingWallet)`

File: universal-login-react/src/playground/CreateRandomInstance.tsx [hunk 5]
```diff
@@ -16,7 +16,9 @@ export const CreateRandomInstance = ({walletService}: CreateRandomInstanceProps)
     const randomString = Math.random().toString(36).substring(7);
     const name = `${randomString}.mylogin.eth`;
     setEnsName(name);
-    const {waitForBalance, contractAddress} = await walletService.createFutureWallet(name);
+    const futureWallet = await walletService.createWallet(name);
```
Comment (xitronix) [inline universal-login-react/src/playground/CreateRandomInstance.tsx:19]: can use createFutureWallet here

**Pull request merged**